transformer_engine/common/include/transformer_engine/fused_attn.h

```diff
@@ -0,0 +1,238 @@
+/*************************************************************************
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+#ifndef TRANSFORMER_ENGINE_FUSED_ATTN_FP8_H_
+#define TRANSFORMER_ENGINE_FUSED_ATTN_FP8_H_
+
+#include "transformer_engine.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum NVTE_QKV_Layout {
+    NVTE_NOT_INTERLEAVED = 0,  /*!< separate Q, K, V matrices */
+    NVTE_QKV_INTERLEAVED = 1,  /*!< packed QKV: [total_seqs, 3, num_heads, head_dim] */
+    NVTE_KV_INTERLEAVED = 2  /*!< Q, packed KV; */
+                        /*!< Q: [total_seqs_q, num_heads, head_dim] */
+                        /*!< KV: [total_seqs_kv, 2, num_heads, head_dim] */
+};
+
+enum NVTE_Bias_Type {
+    NVTE_NO_BIAS = 0,  /*!< no bias */
+    NVTE_PRE_SCALE_BIAS = 1,  /*!< bias before scale */
+    NVTE_POST_SCALE_BIAS = 2  /*!< bias after scale */
+};
+
+enum NVTE_Mask_Type {
+    NVTE_PADDING_MASK = 0,  /*!< padding attention mask */
+    NVTE_CAUSAL_MASK = 1,  /*!< causal attention mask */
+    NVTE_NO_MASK = 2  /*!< no masking */
+};
+
+/*! \brief Compute dot product attention with packed QKV input.
+ *
+ * Computes:
+ *  - P = Q * K.T + Bias
+ *  - S = ScaleMaskSoftmax(P)
+ *  - D = Dropout(S)
+ *  - O = D * V.T
+ *
+ *  \param[in]     QKV                   The QKV tensor in packed format,
+ *                                       [total_seqs, 3, num_heads, head_dim], e.g.
```

Comment:
Hmm, that format explanation should be in the `NVTE_QKV_Layout` instead, right? Here we should just say what we support (probably just below the pseudocode we should have a table that says which combinations of parameters we support currently).

Reply:
/te-ci

Reply:
Done.